app/models/partnership_csv_upload.rb

```diff
@@ -17,7 +17,7 @@ def invalid_schools
   def valid_schools
     invalid_urns = invalid_schools.map { |error| error[:urn] }
     valid_urns = urns - invalid_urns
-    School.where(urn: valid_urns)
+    valid_urns.map { |urn| School.find_by(urn: urn) } # We need to preserve order uploaded
```

Comment:
Ech, how about
```
schools = School.where(urn: valid_urns)
valid_urns.map { |urn| schools.find { |school| school.urn == urn} }
```
to reduce n+1 queries here?

Reply:
They're all cached by this point from `find_school_errors` anyway, don't think it will make much difference

Reply:
Is Rails clever enough to know that we loaded them before?

We don't operate on a scope we got from before, we operate on `School` :/